src/components/Survey.tsx

```
@@ -51,47 +54,140 @@ const groupedInterests = interests.reduce(
+    } else {
+      onValidate(false);
+    }
+  };
+
+  useEffect(() => {
+  
+    setSurveyForm({
+      interestFields: Array.from(checkBox.entries())
+        .filter(([_,isChecked]) => isChecked)
+        .map(([field]) => field as InterestField),
+      interestEtc: Object.fromEntries(interestEtcField)  as Record<InterestField, string>,
```

Biome (1.9.4)

[error] This hook does not specify all of its dependencies: validateSurveyForm

This dependency is not specified in the hook dependency list.


(lint/correctness/useExhaustiveDependencies)